core/cloudflow-flink/src/main/scala/cloudflow/flink/FlinkStreamlet.scala

```diff
@@ -167,12 +167,24 @@ abstract class FlinkStreamlet extends Streamlet[FlinkStreamletContext] with Seri
       StreamExecutionEnvironment.getExecutionEnvironment
     }
 
-    if (!env.getCheckpointConfig.isCheckpointingEnabled()) {
+    if (!env.getCheckpointConfig.isCheckpointingEnabled() && isDefaultCheckpointingEnabled(config, streamlet)) {
```

Comment:
How it will look like, from the user's perspective, to explicitly disable: `isCheckpointingEnabled` ?

Reply:
I'll add documentation

Reply:
We have found ( I guess) that we could not specifically disable checkpointing through flink config. setting a number lower than 10 (where -1 means off) was not possible. So they can't really disable through Flink. (I'm not sure if I am representing your comment @andreaTP but I read into it like that)

Reply:
What is happening here is that if they have not set any value for checkpointing interval, isCheckpointingEnabled is false.

Reply:
got it now! Thanks for the explanation! 👍 
Should we add a mention to this configuration in the relevant doc?

Reply:
Always good to add detail, and we likely need to document this on the flinkside, though I'm not sure how many will want to disable checkpointing, it is basically what you use flink for, to have stateful processing, IMHO.